reporter/reporter.go

```diff
@@ -245,20 +245,19 @@ func (t *reporter) reportDerivedMetrics(span tracer.RawSpan) {
 	replaceTag(tags, "application", appName, appFound)
 	replaceTag(tags, "service", serviceName, svcFound)
 
-	if len(t.redMetricsCustomTagKeys) > 0 {
-		redMetricsCustomTags := make(map[string]string)
-		customTagMatch := false
-		for _, key := range t.redMetricsCustomTagKeys {
-			if value, found := getAppTag(key, "", span.Tags); found {
-				tags[key] = value
-				redMetricsCustomTags[key] = value
-				customTagMatch = true
-			}
-		}
-		if customTagMatch {
-			t.heartbeater.AddCustomTags(redMetricsCustomTags)
+	t.redMetricsCustomTagKeys = append(t.redMetricsCustomTagKeys, string(ext.SpanKind))
```

Comment:
We don't want to be appending this every time we report derived metrics. Should append this just once?

Reply:
Thanks for pointing out, I've changed the variable for a set instead.

Reply:
I notice that the function `RedMetricsCustomTagKeys(redMetricsCustomTagKeys []string)` is never called, should we remove it or did I miss its usage from somewhere else?

Reply:
@yzheqing added a comment else where about that function. The constructor takes an optional list of "setters" at the end. This function being one of them. It is used by users when constructing the reporter.

Reply:
@vikramraman thanks for the clarification! added back the function